code/modules/admin/verbs/freeze.dm

```diff
@@ -26,32 +26,35 @@ var/global/list/frozen_mob_list = list()
 /mob/living/var/frozen = null //used for preventing attacks on admin-frozen mobs
 /mob/living/var/admin_prev_sleeping = 0 //used for keeping track of previous sleeping value with admin freeze
 
-/mob/living/proc/admin_Freeze(var/client/admin, skip_overlays = FALSE)
+/mob/living/proc/admin_Freeze(client/admin, skip_overlays = FALSE)
 	if(istype(admin))
 		to_chat(src, "<b><font color= red>You have been frozen by [admin]</b></font>")
 		message_admins("<span class='notice'>[key_name_admin(admin)]</span> froze [key_name_admin(src)]")
 		log_admin("[key_name(admin)] froze [key_name(src)]")
 
 	var/obj/effect/overlay/adminoverlay/AO = new
 	if(skip_overlays)
-		src.overlays += AO
+		overlays += AO
 
-	anchored = 1
-	frozen = AO
+	anchored = TRUE
+	canmove = FALSE
```

Comment:
Does `update_canmove` check for the value of `frozen`? If it doesn't, any other effect that would call `update_canmove` could possibly set this back to `TRUE`.

Reply:
It doesn't check for frozen specifically. However `canmove` will ultimately be set to `FALSE` if the person is unconscious. And because they're asleep as long as the admin freeze is in effect, `canmove` will never be `TRUE`. 

Setting the frozen person's canmove is probably uneccessary now that I think about it. At the time I thought it would be extra security but I think having their sleeping var unable to be changed is enough.